swift_browser_ui_frontend/src/common/api.js

```diff
@@ -110,39 +89,55 @@ export async function updateBucketMeta(
   return ret;
 }
 
-export async function getObjects(container, signal) {
-  // Fetch objects contained in a container from the API for the user
-  // that's currently logged in.
-  let objUrl = new URL("/api/bucket/objects", document.location.origin);
-  // Search parameter named bucket to avoid changing the API after changing
-  // over from S3 to Swift
-  objUrl.searchParams.append("bucket", container);
-  let objects = fetch(
-    objUrl, { method: "GET", credentials: "same-origin", signal },
-  ).then(
-    function (resp) { return resp.json(); },
-  ).then(
-    function (ret) {
-      for (let i = 0; i < ret.length; i++) {
-        ret[i]["url"] = (
-          "/api/object/dload?bucket=" + container +
-          "&objkey=" + ret[i]["name"]
-        );
-      }
-      return ret;
-    },
+export async function getObjects(
+  project,
+  container,
+  marker = "",
```

Comment:
What is the marker?

Reply:
It's for the Openstack API paging – the actual API can only provide 10k items at once per call, the marker is the latest item so the API knows where to continue. The library provided by Openstack does this by spinning up a request for every page in its own thread, leading to these kinds of long calls taking a lot of time to complete – so now we're doing it in the front-end.